registry/event/event_publishing_service_discovery.go

```diff
@@ -34,6 +39,7 @@ import (
 // Publish some event about service discovery
 type EventPublishingServiceDiscovery struct {
 	serviceDiscovery registry.ServiceDiscovery
+	once             sync.Once
```

Comment:
u add this var just for `epsd.serviceDiscovery.Initialize`. so u may need to change its name which is not so common, IMO.